### PR TITLE
vcpkg port 1.8 fixes

### PR DESCRIPTION
**Pull request by trueqbit**
In order to [update](https://github.com/microsoft/vcpkg/pull/28988) vcpkg's sqlite-orm port, a few things needed to be addressed:
* pinned appveyor test environment to vcpkg 2023.01.09
* made `type_is_nullable<>` a specializable struct again
* examples are now fully linkable again: chrono_binding.cpp, nullable_enum_binding.cpp, pointer_passing_interface.cpp (those examples wrongly commented out the `main()` function)
* updated unit test framework to Catch2 v3 (#1114)
* updated clang-format lint action on github
* removed dangling submodule reference (#1107)
* made all examples compilable with Visual C++ 2015 Update 3

Update (2023-01-15):
Currently unit tests fail with Visual C++ 2015 Update 3 as the minimal C++14 base compiler, as Catch2 v3 now requires a fully compliant C++14 compiler. Let's see what the decision is on the [feature request] (https://github.com/catchorg/Catch2/issues/2624) I made.

Update (2023-01-24):
Now that Catch2 v3 requires a conforming C++14 compiler, legacy compilers like Visual C++ 2015 Update 3 can't be used anymore for compiling the unit tests. The library itself and its examples are still usable with legacy compilers.

### CODE REVIEW

File: examples/chrono_binding.cpp [hunk 1]
```diff
@@ -6,6 +6,10 @@
 #include <format>
 
 #if __cpp_lib_chrono >= 201907L && __cpp_lib_format >= 201907L
+#define ENABLE_THIS_EXAMPLE
```
Comment (fnc12) [inline examples/chrono_binding.cpp:9]: what is the main different of this file's modification? it still doesn't work if `!(__cpp_lib_chrono >= 201907L && __cpp_lib_format >= 201907L)` and reverse

Comment (trueqbit) [reply examples/chrono_binding.cpp:3]: We have to provide `int main()`, and `ENABLE_THIS_EXAMPLE` conditionally compiles main()'s body, see code further down.

**Review by fnc12 — APPROVED** [review]

**Review by fnc12 — APPROVED** [review]

File: examples/blob_binding.cpp [hunk 2]
```diff
@@ -15,18 +15,28 @@ struct Rect {
     int y = 0;
     int width = 0;
     int height = 0;
+
+#ifndef SQLITE_ORM_AGGREGATE_NSDMI_SUPPORTED
```
[the remark this answers is not in the document]
Comment (trueqbit) [reply examples/blob_binding.cpp:19]: Unfortunately, NSDMI is a core capability of a compiler, so you have to compile it with a compiler that does not have it.
Anyway, I suggest that the v1.8 band is the last one in which we support compilers pre-C++14 compilers. As you may have noticed, Martin Hořeňovský, the maintainer of Catch2, is not interested in maintaining compatibility. So the problem is that it is becoming more and more expensive to maintain compatibility on our side.

[the remark this answers is not in the document]
Comment (fnc12) [reply examples/blob_binding.cpp:19]: got it. Yes I agree with dropping pre-C++14 compilers support from next releases

**Pull request merged**